algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+        if args.resume_latest_config:
+            print('args.resume_latest_config is set on')
+            # read last_config.json if it exists, else create it below
+            if os.path.exists(args.last_config_resume_path):
+                print('Folder for this args.experiment_id "{}" and latest_config file existed'
+                      .format(args.experiment_id))
+                with open(args.last_config_resume_path, 'r') as f:
+                    print('Loading config_dict from file: {}'.format(args.last_config_resume_path))
+                    print('Therefore changes to args.config_file_name won\'t affect the env.\n'
+                          'Set to args.resume_latest_config to 0 if you don\'t want this')
+                    args.config_dict = json.load(f)
+        else:
+            print('Loading most params from args.config-file-name')
 
-    env.close()  # above env initialisation was only to find certain params needed
+        # random actions on reset to encourage robustness
+        args.config_dict['num_random_actions_at_init'] = args.num_random_actions_at_init
+        args.config_dict['max_episode_length'] = args.max_episode_length
+        # use given config file to start config and then allow config_dict to overwrite values
+        config_file_dir_path = os.path.abspath(os.path.join(__file__, '../../..', 'gym_ai2thor',
+                                                            'config_files'))
+        args.config_file_path = os.path.join(config_file_dir_path, args.config_file_name)
+        env = AI2ThorEnv(config_file=args.config_file_path, config_dict=args.config_dict,
+                         seed=args.seed)  # todo decide
+        args.resolution = (env.config['resolution'][0], env.config['resolution'][1])
 
+    # Create shared model
+    if env.task.task_has_language_instructions:
+        # environment will return natural language sentence as part of state so process it with
+        # Gated Attention (GA) variant of A3C
+        shared_model = A3C_LSTM_GA(env.observation_space.shape[0], env.action_space.n,
+                                   args.resolution, len(env.task.word_to_idx),
+                                   args.max_episode_length)
+    else:
+        shared_model = ActorCritic(env.observation_space.shape[0], env.action_space.n,
+                                   args.resolution)
+    shared_model.share_memory()
+
+    # Create optimizer
     if args.no_shared:
         optimizer = None
     else:
         optimizer = my_optim.SharedAdam(shared_model.parameters(), lr=args.lr)
         optimizer.share_memory()
 
-    processes = []
+    env.close()  # above env initialisation was only to find certain params needed for models
 
-    counter = mp.Value('i', 0)
-    lock = mp.Lock()
+    # Checkpoint creation/loading below
+    checkpoint_counter = False
+    if not os.path.exists(args.checkpoint_path):
+        print('Tensorboard created experiment folder: {} and checkpoint folder '
```

Comment:
```suggestion
        print('Creating tensorboard experiment folder: {} and checkpoint folder '
```

Reply:
If an exception comes out you know that there was a problem creating the folder without even looking at the code